R/database.R

```diff
@@ -1715,98 +1715,154 @@ get_sample_cn_segments = function(this_sample_id,
 
 #' Retrieve all copy number segments from the GAMBL database that overlap with a single genomic coordinate range.
 #'
-#' @param chromosome The chromosome you are restricting to.
-#' @param qstart Start coordinate of the range you are restricting to.
-#' @param qend End coordinate of the range you are restricting to.
-#' @param region Region formatted like chrX:1234-5678 instead of specifying chromosome, start and end separately.
-#' @param with_chr_prefix Prepend all chromosome names with chr (required by some downstream analyses).
+#' @param region Region formatted like chrX:1234-5678 or X:1234-56789.
+#' @param chromosome The chromosome you are restricting to. Required parameter if region is not specified.
+#' @param qstart Start coordinate of the range you are restricting to. Required parameter if region is not specified.
+#' @param qend End coordinate of the range you are restricting to. Required parameter if region is not specified.
+#' @param projection Selected genome projection for returned Cn segments.
+#' @param this_seq_type Seq type for returned Cn segments. Currently, only genome is supported. Capture samples will be added once processed through CNV protocols.
+#' @param remove_chr_prefix Prepend all chromosome names with chr (required by some downstream analyses). Default is FALSE. 
```

Comment:
Maybe the description should be not "Prepend ...", but rather "Remove ..."? Otherwise does not sound to match the parameter name

Reply:
Yeah I was confused about that too. with_chr_prefix seemed reasonable to me because it means the user wants the result to have chr prefix there. Prepend/remove implies an action will be done. The "with" naming implies it will be done, when necessary, if requested (if that makes any sense)

Reply:
Those are good points, I will revert the name change here. However, I do still think the old name is somewhat misleading. This is because this parameter only has one condition in the code, if set to FALSE (default) chr prefixes will be removed. At least to me, this implies that if set to TRUE, chr prefixes should be available. But that's not how the function operates. I can easily update this so that if set to TRUE, "chr" will be added (if not already there, i.e if returned segments are in respect to hg38). Does that make sense?

Reply:
Yes that makes sense. Perhaps this functionality just wasn't added when this function was updated to support hg38?